### ISSUE TITLE
Database rebuild/recovery fails with "os error 3" (path not found) under a long .beads directory path on Windows, and the fallback restore also fails

### ISSUE THREAD

**Issue by cderv** [Dicklesworthstone/beads_rust#462]
**Summary**

Under a `.beads` directory path long enough that some of the temp/sidecar filenames `br` constructs during a rebuild push the full path over Windows' 260-character limit, the rebuild fails with an I/O error, and the fallback restore-from-backup that's supposed to recover the original database *also* fails with the same error — leaving the workspace in a state I haven't found a way out of short of shortening the path and starting over.

**Observed**

With a `.beads` directory path around ~200-210 characters (well short of 260 on its own), a fresh JSONL rebuild produced:

```
Error: Automatic database recovery failed (I/O error: The specified path could not be found. (os error 3)); original database restore from '<...>\.beads\.br_recovery' also failed: I/O error: ... (os error 3)
```

I also separately hit a related failure — a false-positive path-validation rejection — at a slightly different path length, where a file that is legitimately inside `.beads/` gets rejected as being outside it:

```
Error: Configuration error: Path '<...>\.beads\issues.jsonl' is outside the beads directory '<...>\.beads'
```

Both only show up once the relevant absolute path length gets into the 250-290 character range; a short path (e.g. directly under a drive root) is clean.

**Expected**

Either the rebuild/recovery succeeds regardless of path depth, or at minimum a failure during rebuild shouldn't also take out the fallback restore path, since that appears to leave the workspace without an automatic recovery option.

**Reproduction**

I didn't reproduce the exact error text from a real report I was looking into (which was about the VACUUM-INTO compaction step specifically), but I could reproduce the same "os error 3, and the fallback also fails" shape by just picking a directory name long enough:

```bash
mkdir -p "/some/path/of/roughly/two/hundred/characters/.beads"
cd "/some/path/of/roughly/two/hundred/characters"
# any small JSONL corpus is enough
printf '{"id":"repro-0001","title":"t","description":"d","status":"open","priority":2,"issue_type":"task","created_at":"2026-08-01T10:00:00Z","created_by":"repro","updated_at":"2026-08-20T10:00:00Z","source_repo":"repro","compaction_level":0,"original_size":0,"dependencies":[]}\n' > .beads/issues.jsonl
br list --limit 1
```

The exact directory-name length that triggers it seems to depend on how long the generated temp/sidecar filenames end up being for that run (they include a PID and a timestamp), so it may take a bit of adjusting to land in the failing range on a different machine.

**Environment**

- Windows 11, `br 0.5.3`

**Hypothesis (unconfirmed)**

My reading of the source is that path canonicalization for validation (via the `dunce` crate) and the temp/sidecar filenames used during compaction (`.{stem}.vacuum.{pid}.{timestamp}.tmp`, plus suffixes like `-wal`, `-wal-cert`, `-wal-cert-head`) don't use Windows' `\\?\` extended-length path prefix anywhere, as far as I can find — `dunce` is only ever used to *strip* that prefix for display/comparison, never to opt into long-path support. If that's right, once the full path (directory + generated filename + sidecar suffix) crosses 260 characters, some of the underlying file operations would hit the classic Windows `MAX_PATH` wall, and — separately — comparing a canonicalized path that happened to stay under the limit against one that didn't could behave inconsistently, since `dunce::canonicalize` may return one in verbatim (`\\?\`) form and the other not. I could be misreading the canonicalization behavior here, so please correct me if there's already handling for this I'm missing.

**Comment by Dicklesworthstone** [comment]
Fixed on main and verified on Windows 11 with a native build of main, at the two path lengths that matter (workspace cwd of 234 and 245 characters, so the sidecar/temp names cross 260): `br list --limit 1` (fresh JSONL rebuild), `br update --notes`, `br sync --flush-only`, `br sync --import-only --rebuild`, and `br doctor` all succeed; before the fix the same script reproduced your exact `os error 3` on rebuild plus the failing fallback restore.

Two independent causes, both yours to spot:

1. `\\?\` vs plain spellings. `dunce::canonicalize` only strips the verbatim prefix when the result is a legal legacy path, so once a member of `.beads/` crossed `MAX_PATH` it canonicalized to `\\?\C:\...` while the directory stayed `C:\...`, and the byte-wise `starts_with` in sync path validation called the file external. Prefix checks in that validator now compare the verbatim-stripped spellings (4b28ee31).
2. br's own no-replace renames on Windows go straight to `MoveFileExW` (fresh-database install, recovery staging, base-snapshot publication). Rust's std upgrades long paths to the extended-length form transparently, but a raw Win32 call gets the exact bytes, hence `ERROR_PATH_NOT_FOUND`. Those calls now use the extended-length spelling once the absolute path nears `MAX_PATH` (2551c8b2). The engine itself (FrankenSQLite, pure Rust on std) already handled long paths.

This ships in the next release; if a different long-path seam shows up please reopen with the trace.

**Comment by Dicklesworthstone** [comment]
Shipped in **v0.5.4**: https://github.com/Dicklesworthstone/beads_rust/releases/tag/v0.5.4 (all 7 platforms — linux amd64/arm64/musl_amd64/musl_arm64, darwin amd64/arm64, windows amd64; binaries signed, SHA256SUMS + per-asset checksums, verified `br --version` = 0.5.4).
